nucypher/characters/lawful.py

```diff
@@ -238,71 +248,73 @@ def create_policy(self, bob: "Bob", label: bytes, **policy_params):
-            raise ValueError("Policy end time must be specified as 'expiration' or 'payment_periods', got neither.")
+        if not duration and not expiration:
+            raise ValueError("Policy end time must be specified as 'expiration' or 'duration', got neither.")
 
         # Merge injected and default params.
         threshold = threshold or self.threshold
         shares = shares or self.shares
-        base_payload = dict(threshold=threshold, shares=shares, expiration=expiration)
-
-        if self.federated_only:
-            if not expiration:
-                raise TypeError("For a federated policy, you must specify expiration (payment_periods don't count).")
-            if expiration <= maya.now():
-                raise ValueError(f'Expiration must be in the future ({expiration}).')
-        else:
-            blocktime = maya.MayaDT(self.policy_agent.blockchain.get_blocktime())
-            if expiration and (expiration <= blocktime):
-                raise ValueError(f'Expiration must be in the future ({expiration} is earlier than blocktime {blocktime}).')
-
-            # Calculate Policy Rate and Value
-            payload = super().generate_policy_parameters(number_of_ursulas=shares,
-                                                         payment_periods=payment_periods,
-                                                         expiration=expiration,
-                                                         *args, **kwargs)
-            base_payload.update(payload)
+        duration = duration or self.duration
+        rate = rate if rate is not None else self.rate  # TODO conflict with CLI default value, see #1709
+        payment_method = payment_method or self.payment_method
+
+        # Calculate Policy Rate, Duration, and Value
+        quote = self.payment_method.quote(
+            shares=shares,
+            duration=duration,
+            commencement=commencement.epoch if commencement else None,
+            expiration=expiration.epoch if expiration else None,
+            rate=rate,
+            value=value
+        )
 
-        return base_payload
+        params = dict(
+            payment_method=payment_method,
+            threshold=threshold,
+            shares=shares,
+            duration=quote.duration,
+            commencement=quote.commencement,
+            expiration=quote.expiration,
+            rate=quote.rate,
+            value=quote.value
+        )
+        return params
 
     def _check_grant_requirements(self, policy):
         """Called immediately before granting."""
-
         # TODO: Remove when the time is right.
-        if policy.expiration > END_OF_POLICIES_PROBATIONARY_PERIOD:
-            raise self.ActorError(f"The requested duration for this policy (until {policy.expiration}) exceeds the "
-                                  f"probationary period ({END_OF_POLICIES_PROBATIONARY_PERIOD}).")
+        # from nucypher.config.constants import END_OF_POLICIES_PROBATIONARY_PERIOD
```

Comment:
Question: what about policies that exceed the unbonding time (2 months)? (@cygnusv )

Reply:
Included a todo referencing this point.  This is a great entry point to enforce any kind of maximum policy duration.